src/Microsoft.AspNetCore.Mvc.Razor.Extensions/InjectDirective.cs

```diff
@@ -11,7 +11,10 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
 {
     public static class InjectDirective
     {
-        public static readonly DirectiveDescriptor Directive = DirectiveDescriptorBuilder.Create("inject").AddType().AddMember().Build();
+        public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
+            "inject",
+            DirectiveDescriptorKind.SingleLine,
```

Comment:
This doesn't feel good, also we have `DirectiveDescriptorKind` and `DirectiveTokenKind`. We should be consistent with the use of `Descriptor`.